tabled/examples/interactive.rs

```diff
@@ -0,0 +1,135 @@
+use std::{thread::sleep, time::Duration};
+
+use tabled::{
+    settings::{
+        object::{ObjectIterator, Rows},
+        style::BorderColor,
+        themes::Colorization,
+        Color, Style,
+    },
+    Table,
+};
+use tabled_derive::Tabled;
+
+#[derive(Tabled, Clone, Debug)]
+struct Item {
+    name: &'static str,
+    category: &'static str,
+    value: f64,
+}
+
+const CLEAR: &str = "\u{1b}[2J";
+
+type Step = Box<dyn Fn(&mut Table) -> (u64, &mut Table)>;
```

Comment:
Ahh kind of hard to understand right?)

Probably will need to refactor it afterall

Reply:
I agree. I could put the steps in its own struct to try and make it a little bit cleaner if that's what you meant.

Like:
```rust
struct StepResult {
  // could have `Duration` directly here instead of `u64` milliseconds
  duration: u64,
  table: &mut Table,
}

struct Step {
  action: Box<dyn Fn(&mut Table) -> StepResult>
}
```

Not sure if this is syntactically correct, I haven't tried it.